components/application/activities.js

```diff
@@ -0,0 +1,74 @@
+const transform = (current, parent) => {
+  const result = { ...current, parent }
+
+  result.test = (str) =>
+    (parent?.test?.call(str) || true) && current.test.test(str)
+
+  if (parent?.path) result.path = [parent.path, current.path].join('/')
+
+  if (current.children)
+    result.children = current.children.map((child) => transform(child, result))
+
+  return result
+}
+
+class RouteConfig {
+  constructor(init) {
+    this.routes = init
+      .map((route) => transform(route, null))
+      .flatMap((route) => [...(route.children || []), route])
+  }
+
+  get(name) {
+    return this.routes.find((route) => route.path === name)
+  }
+
+  find(url) {
+    return this.routes.find((route) => route.test(url))
+  }
+}
+
+const config = new RouteConfig([
+  {
+    test: /\/overview/,
+    path: 'overview',
+    icon: 'view-dashboard',
+  },
```

Comment:
Doing so I come to the idea that we should use some library for route parsing or define our routes in a **much more** simple way.

Reply:
Routes will need to refactor if we want to show a warning icon in the menu near to issues.

Reply:
I am considering inlining everything into JSX without this configuration.

I wonder if you have any other idea?